src/playwright/transformPlaywright.ts

```diff
@@ -20,17 +20,25 @@ export const testPrefixer = template(
       const testFn = async() => {
         const context = { id: %%id%%, title: %%title%%, name: %%name%% };
 
-        page.on('pageerror', (err) => {
+        const pageErrorListener = (err) => {
           page.evaluate(({ id, err }) => __throwError(id, err), { id: %%id%%, err: err.message });
-        });
+        };
+  
+        page.on('pageerror', pageErrorListener);
 
         if(globalThis.__sbPreRender) {
           await globalThis.__sbPreRender(page, context);
         }
 
-        const result = await page.evaluate(({ id, hasPlayFn }) => __test(id, hasPlayFn), {
-          id: %%id%%,
-        });
+        let result;
+        try {
+          result = await page.evaluate(({ id, hasPlayFn }) => __test(id, hasPlayFn), {
+            id: %%id%%,
+          });
+        } catch (error) {
+          throw error;
+          result = null;
+        }
```

Comment:
If we're just catching and throwing the error, does it make sense to wrap it in a try catch? in this case, `result = null` will never be called because the error is thrown before that line